testbed/lib/infra-stack.ts

```diff
@@ -0,0 +1,194 @@
+    Tags.of(sg).add('kit.sh/stack', super.stackName)
+    Tags.of(vpc).add('kit.sh/stack', super.stackName)
+
+    // Select the private subnets created in our VPC and place our worker nodes there
+    const privateSubnets = vpc.selectSubnets({
+      subnetType: ec2.SubnetType.PRIVATE_WITH_NAT,
+    });
+
+    const ng = cluster.addNodegroupCapacity('WorkerNodeGroup', {
+      subnets: privateSubnets,
+      nodeRole: workerRole,
+      minSize: 3,
+      maxSize: 3,
+      instanceTypes: [
+        new ec2.InstanceType('m5.large'),
+        new ec2.InstanceType('m5a.large'),
+        new ec2.InstanceType('m6i.large'),
+        new ec2.InstanceType('m6a.large'),
+        new ec2.InstanceType('t3.large'),
+        new ec2.InstanceType('t3a.large'),
+        new ec2.InstanceType('c5.large'),
+        new ec2.InstanceType('c5a.large'),
+        new ec2.InstanceType('c6i.large'),
+      ],
+      tags: {
+        "kit.sh/stack": super.stackName,
+      },
+      taints: [
+        {
+          effect: eks.TaintEffect.NO_SCHEDULE,
+          key: 'CriticalAddonsOnly',
+          value: 'true',
+        },
+      ],
+    });
+
+    // Setup Tekton test permissions
+
+    const ns = cluster.addManifest('tekton-tests-ns', {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+          name: testNS
+      }
+    });
+
+    const sa = cluster.addServiceAccount('tekton-tests-sa', {
+        name: testSA,
+        namespace: testNS
+    })
+    sa.node.addDependency(ns)
+    sa.role.attachInlinePolicy(new iam.Policy(this, 'tekton-tests-policy', {
+        statements: [
+            new iam.PolicyStatement({
+                resources: ['*'],
+                actions: [
+                    // Write Operations
+                    "ec2:*",
+                    "cloudformation:*",
+                    "iam:*",
+                    "ssm:GetParameter",
+                    "eks:*",
+                    "pricing:GetProducts",
+                ],
+            }),
+        ],
+    }));
+
+    cluster.awsAuth.addRoleMapping(sa.role, {
+        username: 'system:node:{{EC2PrivateDNSName}}',
+        groups: ['system:bootstrappers', 'system:nodes']
+    })
+
+    // Install cluster add-ons for the host cluster
+
+    new AWSEBSCSIDriver(this, 'AWSEBSCSIDriver', {
+      cluster: cluster,
+      namespace: 'aws-ebs-csi-driver',
+      version: 'v1.9.0',
+      chartVersion: 'v2.8.1',
+    }).node.addDependency(cluster);
+
+    new FluxV2(this, 'Flux', {
+      cluster: cluster,
+      namespace: 'flux-system',
+      repoUrl: repoUrl,
+      repoBranch: repoBranch,
+      repoPath: repoPath,
+      testRepoName: testRepoName,
+      testRepoUrl: testRepoUrl,
+      testRepoBranch: testRepoBranch,
+      testRepoPath: testRepoPath,
+    }).node.addDependency(cluster);
+
+    new AWSLoadBalancerController(this, 'AWSLoadBalancerController', {
+      cluster: cluster,
+      namespace: 'aws-load-balancer-controller',
```

Comment:
likewise.

Reply:
yep, but.... to my knowledge that's arbitrary just because it's easiest. Seems better to run in separate namespaces, if for nothing else, ease of seeing an application resources.